### DISCUSSION TITLE
How to create ISMRMRD dataset file for 3D data.

### DISCUSSION THREAD

**Discussion by Diya95** [gadgetron/gadgetron#1184] [Q&A]
I have a 3D raw dataset with matrix size 256 x 256 x 96 .I am familiar with creating ISMRMRD dataset for 2D data as suggested here : [https://github.com/ismrmrd/ismrmrd/blob/master/examples/matlab/create_dataset.m](url)

Now I need to know 
1) How the above code changes for creating ISMRMRD for 3D dataset.
2) How to fill the kspace_encoding_step_2 in ISMRMRD header.
3) How to fill the Z dimension values.

**Comment by hansenms** [comment]
You should use the kspace_encoding_step_2 in the same way that you use kspace_encoding_step_1, but for the kz dimension instead. I am not sure I fully get the question? It just adds another encoding dimension that you fill limits for in the header and you increment the index on individual lines. 

What have you tried so far and in what ways is it not working?
